### ISSUE TITLE
Rationalize our use of hubs and repos

### ISSUE THREAD

**Issue by henrynash** [appsody/stacks#265]
Today, we present a confusing picture about what is a repo and what is a hub (that contains repos). For clarity, when I use "repo" in this issue, I explicitly mean a "repo that holds stacks". Examples of this confusion:

1) According to the CLI & our docs, "appsodyhub" is a repo (this is in multiple places). However, I believe, in fact, it is a hub that contains three repos (stable, incubating, experimental)
2) When you use 'appsody init' or 'appsody list' it is not only defaulting to the appsodyhub as a location for repos, but also defaulting to the incubating repo. I gather the intention is, in the future, that this default will switch to stable. I would suggest that changing such a default will break people (e.g. what if they have tooling wrapped around our CLI?).
3) The CLI can also be used to reference an explicit local repo (e.g. 'appsody init mylocalrepo/mystack"). However, I can't do any of:  'appsody init incubating/stackname', 'appsody list incubating", 'appsody list appsodyhub/incubating", etc.

Given that our hope is for 100s of stacks to be created (by us, clients, GSIs etc.), we need to be clear how this is going to work in the future - or we will create significant problems for ourselves. We need to clear on our conceptual strategy, i.e.: Is it the goal that our repos are the default public repos? The only recognised public repos? Or should there be a hub that contains a registry of all public repos so people can find them? If so, is that appsodyhub? How will the Appsody open source community manage these repos?  Why can't I use a docker registry to hold stacks? etc. etc.

This is obviously quite a complex issue, with many ramifications - so I am raising it here to initiate the conversation. I will post some of my own ideas to this thread.

**Comment by neeraj-laad** [comment]
I agree with the confusion caused by using the term 'appsodyhub' as a repository id. However, I wanted to mention some of the capabilities you listed in point 3 as missing are already there.

`appsody list` - lists all stacks from all repositories

`appsody list <repo-id>` - lists all stacks from the repo: <repo-id>

`appsody init <repo-id>/<stack-id>` - initializes a stack: <stack-id> from repo: <repo-id>

Perhaps we should just call these repos experimental/incubator/stable. And use Appsody hub as a way to refer to the official repos or a UI/control point that allows you to manage these repos and stacks in them.

**Comment by henrynash** [comment]
So yes, I think simply moving to calling our repos stable, incubating and experimental would be a great start - and making sure our CLI is consistent. I realise the capabilities you list are present - but they are not supported consistently, i.e. I can't do something like:

appsody init incubating/nodejs 

or more usefully:

appsody init experimental/python-flask

...since there is no other way of getting at the experimental ones, iunless you build them locally and add it to a local repo. Maybe someone really worked out the naming conventions within appsodyhub, they might be able to great a local repo that pointed at it? All far too much like hard work.
